src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseAllocationPayBand.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderProgramProfile
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderProgramProfilePayBand
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderProgramProfilePayBandId
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.PayBand
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
+import java.time.LocalDate
+
+@DslMarker
+annotation class CourseAllocationPayBandDslMarker
+
+@TestDataDslMarker
+interface CourseAllocationPayBandDsl
+
+@Component
+class CourseAllocationPayBandBuilderRepository(private val payBandRepository: ReferenceCodeRepository<PayBand>) {
+  fun payBand(payBandCode: String): PayBand = payBandRepository.findByIdOrNull(PayBand.pk(payBandCode))!!
+}
+
+@Component
+class CourseAllocationPayBandBuilderFactory(private val repository: CourseAllocationPayBandBuilderRepository) {
```

Comment:
This is an example of a child entity's builder which only needs to know about its own dedicated repository